src/azul/queues.py

```diff
@@ -235,15 +235,19 @@ def wait_to_stabilize(self) -> int:
             logger.info('Message count history (most recent first) is %r.',
                         list(reversed(total_lengths)))
 
-            if len(total_lengths) == total_lengths.maxlen:
-                cummdiff = sum(
-                    abs(first - second)
-                    for first, second in more_itertools.pairwise(total_lengths)
-                )
-                if cummdiff == 0:
-                    final_length = total_lengths[-1]
-                    logger.info('The queues have stabilized.')
-                    break
+            min_iter = 60 // sleep_time
+            total_lengths_len = len(total_lengths)
+            if total_lengths_len >= min_iter:
+                is_zeros = set(list(total_lengths)[-min_iter:]) == {0, }
```

Comment:
Answer the following questions: How many copies of the deque does this create? How many of them are partial? How many collections does this solution involve? List them.

Change this to a solution that does not create any copies. Hint: `islice` & `any`.

Reply:
>Answer the following questions: How many copies of the deque does this create? How many of them are partial? How many collections does this solution involve? List them.

It creates three copies, two are partial and it involves three collections, deque, list and set.